tensorflow/python/ops/random_ops.py

```diff
@@ -291,6 +291,11 @@ def random_uniform(shape,
     if dtype.is_integer:
       raise ValueError("Must specify maxval for integer dtype %r" % dtype)
     maxval = 1
+  if minval >= maxval:
```

Comment:
since these are potentially tensor arguments, you cannot do this comparison in python

Reply:
@cantonios , That's True. How about modifying the class PhiloxRandomOp in [random_op.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/random_op.cc#L64) like below. It may verify other dtypes. Amended code commented as **//added code**. Is this works?

```
template <typename Device, class Distribution>
class PhiloxRandomOp : public OpKernel {
 public:
  typedef typename Distribution::ResultElementType T;
  explicit PhiloxRandomOp(OpKernelConstruction* ctx) : OpKernel(ctx) {
    OP_REQUIRES_OK(ctx, generator_.Init(ctx));
  }

  void Compute(OpKernelContext* ctx) override {
    const Tensor& shape = ctx->input(0);
    const Tensor& minval = ctx->input(1); //added code
    const Tensor& maxval = ctx->input(2); //added code
    Tensor* output;
    OP_REQUIRES_OK(ctx, AllocateOutputWithShape(ctx, shape, 0, &output));
    auto output_flat = output->flat<T>();
    
    Distribution lo = minval.scalar<Distribution>()();//added code
    Distribution hi = maxval.scalar<Distribution>()();//added code
    OP_REQUIRES(
        ctx, lo < hi,
        errors::InvalidArgument("Need minval < maxval, got ", lo, " >= ", hi));//added code
        
    functor::FillPhiloxRandom<Device, Distribution>()(
        ctx, ctx->eigen_device<Device>(), /*key=*/nullptr, /*counter=*/nullptr,
        // Multiplier 256 is the same as in FillPhiloxRandomTask; do not change
        // it just here.
        generator_.ReserveRandomOutputs(output_flat.size(), 256),
        output_flat.data(), output_flat.size(), Distribution());
  }

 private:
  GuardedPhiloxRandom generator_;
};
```

Reply:
No, the min/max are not arguments to the op.

You either need to use something in `check_ops.py`, or you clamp the maximum to the minimum, or you keep the behavior as it is.

Reply:
Changed the validation using `check_ops.assert_less ` as suggested.